src/site/content/en/blog/javascript-this/index.md

```diff
@@ -0,0 +1,356 @@
+---
+title: 'JavaScript: What is the meaning of this?'
+subhead: "Figuring out the value of `this` can be tricky in JavaScript, here's how to do it…"
+description: "Figuring out the value of `this` can be tricky in JavaScript, here's how to do it…"
+authors:
+  - jakearchibald
+date: 2021-02-22
+# updated: 2021-02-19
+hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/TuciUuOQOd3u7uMgDZBi.jpg
+alt: A forest stream with colored fallen leaves.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - javascript
+---
+
+TODO: date & image
+
+JavaScript's `this` is the butt of many jokes, and that's because, well, it's pretty complicated. However, I've seen developers do much-more-complicated and domain-specific things to avoid dealing with this `this`. If you're unsure about `this`, hopefully this will help. This is my `this` guide.
+
+I'm going to start with the most specific situation, and end with the least-specific. This article is kinda like a big `if (…) … else if () … else if (…) …`, so you can go straight to the first section that matches the code you're looking at.
+
+1. [If the function is defined as an arrow function](#arrow-functions)
+1. [Otherwise, if the function/class is called with `new`](#new)
+1. [Otherwise, if the function has a 'bound' `this` value](#bound)
+1. [Otherwise, if `this` is set all call-time](#call-apply)
+1. [Otherwise, if the function is called via a parent object (`parent.func()`)](#object-member)
```

Comment:
Nit: if possible, I'd reorder sections by popularity. For example, I'm sure that `parent.func()` is far more common than e.g. `bound this value` so I'd push it up.

Reply:
@RReverser I don't think that's possible. Right now it's ordered by specificity, and moving `parent.func()` would break that, requiring more rules and exceptions to make sense of it all.